File: db/src/main/java/org/apple/db/user/UserEntity.java
package org.apple.db.user;

import jakarta.persistence.Column;
import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import org.apple.db.BaseEntity;

@Entity
public class UserEntity extends BaseEntity {

    @Column
    private String name;
    @Column
    private String email;
    @Embedded
    private Address address;
}

File: db/src/main/kotlin/org/apple/db/user/UserEntity.kt
package org.apple.db.user

import jakarta.persistence.*
import org.apple.db.Address
import org.apple.db.BaseEntity
import org.apple.db.user.enums.UserStatus

@Entity
@Table(name = "user")
class UserEntity(
    @field:Column(length = 50, nullable = false)
    val name: String,

    @field:Column(length = 100, nullable = false)
    val email: String,

    @field:Column(length = 100, nullable = false)
    val password: String,

    @field:Embedded
    var address: Address?= null,

    @field:Column(length = 50, nullable = false)
    @field:Enumerated(EnumType.STRING)
    var status: UserStatus,
) : BaseEntity()


